pkg/controller/perconaservermongodb/ssl.go

```diff
@@ -42,57 +43,314 @@ func (r *ReconcilePerconaServerMongoDB) reconsileSSL(ctx context.Context, cr *ap
 		return errors.Wrap(errInternalSecret, "get internal SSL secret")
 	}
 
-	err := r.createSSLByCertManager(ctx, cr)
+	ok, err := r.isCertManagerInstalled(ctx, cr.Namespace)
 	if err != nil {
-		logf.FromContext(ctx).Error(err, "issue cert with cert-manager")
+		return errors.Wrap(err, "check cert-manager")
+	}
+	if !ok {
+		if errSecret == nil && errInternalSecret == nil {
+			return nil
+		}
 		err = r.createSSLManually(ctx, cr)
 		if err != nil {
 			return errors.Wrap(err, "create ssl manually")
 		}
+		return nil
+	}
+	err = r.createSSLByCertManager(ctx, cr)
+	if err != nil {
+		return errors.Wrap(err, "create ssl by cert-manager")
 	}
 	return nil
 }
 
+func (r *ReconcilePerconaServerMongoDB) isCertManagerInstalled(ctx context.Context, ns string) (bool, error) {
+	c := r.newCertManagerCtrl(r.client, r.scheme, true)
+	err := c.Check(ctx, r.restConfig, ns)
+	if err != nil {
+		switch err {
+		case tls.ErrCertManagerNotFound:
+			return false, nil
+		case tls.ErrCertManagerNotReady:
+			return true, nil
+		}
+		return false, err
+	}
+	return true, nil
+}
+
 func (r *ReconcilePerconaServerMongoDB) createSSLByCertManager(ctx context.Context, cr *api.PerconaServerMongoDB) error {
-	c := tls.NewCertManagerController(r.client, r.scheme)
+	log := logf.FromContext(ctx).WithName("createSSLByCertManager")
+
+	dryController := r.newCertManagerCtrl(r.client, r.scheme, true)
```

Comment:
why is this named `dryController`? `dry` sounds like it won't do anything but from the code I see it's doing many things